File: backend/LibraryAPI/Models/Book.cs
using System.ComponentModel.DataAnnotations;

namespace LibraryAPI.Models
{
    public class Book
    {
        public int Id { get; set; }
        [Required]
        [MaxLength(200)]

        public string Title {get; set;} = string.Empty;

        [Required]
        [MaxLength(150)]
        public string Author {get; set;} = string.Empty;

        [MaxLength(1000)]
        public string? Description {get; set;} = string.Empty;

        public DateTime CreatedAt {get; set;} = DateTime.UtcNow;
        public DateTime UpdatedAt {get; set;} = DateTime.UtcNow;
    }
}

File: frontend/src/types/book.ts
export interface Book {
  id: number;
  title: string;
  author: string;
  description: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateBookPayload {
  title: string;
  author: string;
  description: string;
}

export interface UpdateBookPayload {
  title: string;
  author: string;
  description: string;
}

export interface ApiError {
  message: string;
}
export interface PagedBooksResponse {
  data: Book[]
  totalCount: number
  totalPages: number
  currentPage: number
  pageSize: number
  hasNextPage: boolean
  hasPreviousPage: boolean
}